### PR TITLE
Fixed issue #1717: Code coverage turned slow after update from 2.7.2 to 2.8.0

### PR DESCRIPTION
**Pull request by derickr**
New algorithm that does less work to speed this up *a lot*. With thanks to @nikic for the idea.

### CODE REVIEW

File: xdebug_code_coverage.c [hunk 1]
```diff
@@ -1056,38 +1056,18 @@ static int prefill_from_function_table(zend_op_array *opa)
...
-
 static int prefill_from_class_table(zend_class_entry *ce)
 {
 	if (ce->type == ZEND_USER_CLASS) {
-		if (!mark_class_as_visited(ce)) {
-			zend_op_array *val;
+		zend_op_array *val;
 
-			xdebug_zend_hash_apply_protection_begin(&ce->function_table);
+		xdebug_zend_hash_apply_protection_begin(&ce->function_table);
```
Comment (nikic) [inline xdebug_code_coverage.c]: Side note: What's the apply protection here for?

Comment (derickr) [reply xdebug_code_coverage.c]: I don't really know to be honest. AFAIK this is only for recursion protection, right?

Comment (nikic) [reply xdebug_code_coverage.c]: That's right. There should be not possibility for recursion here.

**Comment by derickr** [comment]
I'm closing this in favour of #529

**Pull request closed without merging**